Data race in `Worker.Platforms()` causes `panic: runtime error: index out of range`

### Contributing guidelines and issue reporting guide

- [x] I've read the [contributing guidelines](https://github.com/moby/buildkit/blob/master/.github/CONTRIBUTING.md) and wholeheartedly agree. I've also read the [issue reporting guide](https://github.com/moby/buildkit/blob/master/.github/issue_reporting_guide.md).

### Well-formed report checklist

- [x] I have found a bug that the documentation does not mention anything about my problem
- [x] I have found a bug that there are no open or closed issues that are related to my problem
- [x] I have provided version/information about my environment and done my best to provide a reproducer

### Description of bug

### Description

Concurrent gRPC calls to `ListWorkers` trigger a data race in `Worker.Platforms()`, resulting in a panic.

### Stack Trace

```
panic: runtime error: index out of range [81] with length 81

goroutine 1018 [running]:
github.com/moby/buildkit/worker/base.(*Worker).Platforms(...)
        worker/base/worker.go:285
github.com/moby/buildkit/control.(*Controller).ListWorkers(...)
        control/control.go:624
```

### Root Cause

`Worker.Platforms(noCache bool)` at [`worker/base/worker.go:290`](https://github.com/moby/buildkit/blob/master/worker/base/worker.go#L290) has a data race when `noCache=true`:

```go
func (w *Worker) Platforms(noCache bool) []ocispecs.Platform {
    if noCache {
        matchers := make([]platforms.MatchComparer, len(w.WorkerOpt.Platforms))  // (1) read length
        for i, p := range w.WorkerOpt.Platforms {                                 // (2) iterate
            matchers[i] = platforms.Only(p)
        }
        for _, p := range archutil.SupportedPlatforms(noCache) {
            // ...
            if !exists {
                w.WorkerOpt.Platforms = append(w.WorkerOpt.Platforms, p)          // (3) mutate
            }
        }
    }
    return w.WorkerOpt.Platforms
}
```

When two goroutines call `Platforms(true)` concurrently on the same `Worker` instance:

1. **Goroutine A** reads `len(w.WorkerOpt.Platforms)` = 81, creates `matchers` with length 81
2. **Goroutine B** enters the method, iterates through the first loop, and appends new platforms via `w.WorkerOpt.Platforms = append(...)` at step (3), extending the slice
3. Due to the data race (undefined behavior), **Goroutine A**'s `range` at step (2) may observe the extended slice, causing `i` to exceed `matchers` bounds → **panic**

### How to Reproduce

Trigger multiple concurrent `ListWorkers` gRPC calls (e.g., running `docker buildx ls` from multiple terminals simultaneously, or CI systems polling worker status). The race is more likely when `noCache=true` causes `SupportedPlatforms` to return newly detected platforms that trigger `append`.

### Fix

Add a `sync.Mutex` to protect the `Platforms` method:

```diff
 type Worker struct {
 	WorkerOpt
 	CacheMgr        cache.Manager
 	SourceManager   *source.Manager
 	imageWriter     *imageexporter.ImageWriter
 	ImageSource     *containerimage.Source
 	OCILayoutSource *containerimage.Source
 	GitSource       *git.Source
 	HTTPSource      *http.Source
+	platformsMu     sync.Mutex
 }

 func (w *Worker) Platforms(noCache bool) []ocispecs.Platform {
+	w.platformsMu.Lock()
+	defer w.platformsMu.Unlock()
 	if noCache {
```

### Environment

- BuildKit version: (affected versions including master)
- Docker version: 29.5.x
- OS: Linux

---

--- a/worker/base/worker.go
+++ b/worker/base/worker.go
@@ -7,6 +7,7 @@ import (
 	"os"
 	"path/filepath"
 	"slices"
+	"sync"
 	"time"
 
 	"github.com/containerd/containerd/v2/core/content"
@@ -103,6 +104,7 @@ type Worker struct {
 	OCILayoutSource *containerimage.Source
 	GitSource       *git.Source
 	HTTPSource      *http.Source
+	platformsMu     sync.Mutex
 }
 
 // NewWorker instantiates a local worker
@@ -288,6 +290,8 @@ func (w *Worker) Labels() map[string]string {
 }
 
 func (w *Worker) Platforms(noCache bool) []ocispecs.Platform {
+	w.platformsMu.Lock()
+	defer w.platformsMu.Unlock()
 	if noCache {
 		matchers := make([]platforms.MatchComparer, len(w.WorkerOpt.Platforms))
 		for i, p := range w.WorkerOpt.Platforms {

